api/src/main/java/jakarta/data/exceptions/EntityExistsException.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2022,2023 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.exceptions;
+
+/**
+ * Indicates that an entity cannot be inserted into the database
+ * because an entity with same unique identifier already exists in the database.
+ */
+public class EntityExistsException extends DataException {
```

Comment:
Even inserting and updating several NoSQL databases does not necessarily mean it will work as a relational database.
It might go to several databases that will work in the append model as some wide-column and key-value databases, for example, [Apache Cassandra](https://www.oreilly.com/library/view/cassandra-the-definitive/9781491933657/ch09.html#:~:text=Because%20Cassandra%20uses%20an%20append,not%20exist%2C%20Cassandra%20creates%20it.) and som key-values databases.

Reply:
> It might go to several databases that will work in the append model as some wide-column and key-value databases, for example, [Apache Cassandra](https://www.oreilly.com/library/view/cassandra-the-definitive/9781491933657/ch09.html#:~:text=Because%20Cassandra%20uses%20an%20append,not%20exist%2C%20Cassandra%20creates%20it.) and som key-values databases.

Thanks for pointing this out.  Those types of databases don't offer the insert capability that guarantees you caused the entity to be created and so should raise `UnsupportedOperationException` when the insert operation is attempted.  I added commit https://github.com/jakartaee/data/pull/270/commits/94c74b8f7e373ded7ba9ab07b1817c4e6f253245 to explicitly state this in the spec and JavaDoc.